### PR TITLE
feat: added new api hooks and global states

### PR DESCRIPTION
**Pull request by totraev**

### CODE REVIEW

File: src/app/components/NetworkBadge/NetworkBadge.tsx [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
 
-export const NetworkBadge: React.FC<NetworkBadgeProps> = ({
-  isWalletConnected,
-}) => {
   return (
     <div
-      className={`absolute left-2 ${isWalletConnected ? "top-40 md:top-24 lg:top-32" : "top-24 md:top-24 lg:top-32"}`}
+      className={twJoin(
+        `absolute left-2`,
+        Boolean(walletProvider)
```
Comment (gbarkhatov) [inline src/app/components/NetworkBadge/NetworkBadge.tsx:17]: `Boolean` vs `!!`? Later on I suggest sticking to one approach, right now we have 8 `!!` and 6 `Boolean(something)`

**Review by gbarkhatov — APPROVED** [review]

File: src/app/components/Delegations/Delegations.tsx [hunk 2]
```diff
@@ -32,43 +37,35 @@ import {
...
-          delegationsLocalStorage={delegationsLocalStorage}
-          globalParamsVersion={globalParamsVersion}
-          signPsbtTx={signPsbtTx}
-          pushTx={pushTx}
-          queryMeta={queryMeta}
-          getNetworkFees={getNetworkFees}
+          delegationsAPI={delegationsAPI.delegations}
+          globalParamsVersion={currentVersion}
+          signPsbtTx={signPsbtTransaction(btcWallet)}
+          pushTx={btcWallet.pushTx}
```
[the remark this answers is not in the document]
Comment (totraev) [reply src/app/components/Delegations/Delegations.tsx:67]: @jrwbabylonlab correct, all your questions above relate to `Delegation` component, so I do have plans to create a separate ticket for it

File: src/app/hooks/useDelegations.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  const query = useInfiniteQuery({
+    queryKey: [DELEGATIONS_KEY, publicKeyNoCoord],
+    queryFn: ({ pageParam = "" }) =>
+      getDelegations(pageParam, publicKeyNoCoord),
+    getNextPageParam: (lastPage) =>
+      lastPage?.pagination?.next_key !== ""
+        ? lastPage?.pagination?.next_key
+        : null,
+    initialPageParam: "",
+    refetchInterval: 60000, // 1 minute
```
[the remark this answers is not in the document]
Comment (totraev) [reply src/app/hooks/useDelegations.ts:27]: Good catch!

File: src/app/components/Delegations/Delegations.tsx [hunk 4]
```diff
@@ -81,25 +78,21 @@ export const Delegations: React.FC<DelegationsProps> = ({
...
   btcWalletNetwork: networks.Network;
   address: string;
   signPsbtTx: SignPsbtTransaction;
   pushTx: WalletProvider["pushTx"];
-  queryMeta: QueryMeta;
   getNetworkFees: WalletProvider["getNetworkFees"];
   isWalletConnected: boolean;
 }
 
 const DelegationsContent: React.FC<DelegationsContentProps> = ({
```
[the remark this answers is not in the document]
Comment (totraev) [reply src/app/components/Delegations/Delegations.tsx:91]: No, we will refactor it

**Pull request merged**